src/cdk/a11y/focus-trap.md

```diff
@@ -0,0 +1,12 @@
+### FocusTrap
+The `cdkFocusTrap` directive traps Tab key focus within an element. This is intended to be used to
+create components like [modal dialogs](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal).
```

Comment:
For clarity, maybe we should say `intended to be used to make components like modal dialogs more accessible...`. Think it could be confusing to say it's used to create them, since it's not a building tool.